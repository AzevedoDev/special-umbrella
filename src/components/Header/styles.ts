import styled, { css } from 'styled-components'
import { Colors } from 'styles/theme'
import media from 'styled-media-query'

export const Wrapper = styled.main`
  ${({ theme, color }) => css`
    background-color: ${theme.colors[color as Colors]};
    height: 100%;
    width: 100%;
    font-family: ${theme.font.family};
    z-index: ${theme.layers.base};
    ${media.greaterThan('medium')`
     max-width: 61.5rem;
    `}
  `}
`
export const MobileGroup = styled.div`
  display: none;
  ${media.lessThan('medium')`
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
    padding-top: 6.3rem;
    min-height: 45rem;

    padding-bottom: 4.1rem;
  `};
`
export const DesktopGroup = styled.div`
  display: none;

  ${media.greaterThan('medium')`
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    padding-top: 6.3rem;
    padding-left: 9.4rem;
    min-height: 100%;
    max-width: 61.5rem;
    padding-bottom: 9.3rem;
  `}
`
export const TextWrapper = styled.div``
