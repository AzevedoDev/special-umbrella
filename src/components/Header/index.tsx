import * as S from './styles'
import Logo from 'components/Logo'
import Select from 'components/Select'
import Text from 'components/Text'
import { Loteria } from 'graphql/generated/graphql'
import {
  dateFormatter,
  removeAccents,
  removeSlashesAndEmptySpaces
} from 'utils/helper'

export type HeaderProps = {
  title?: string
  concourseNumber?: string | null
  loterias: Loteria[]
  isMobile?: boolean
  concourseData?: string | null
}

const Header = ({
  title = 'mega-sena',
  concourseNumber = '0',
  loterias,
  isMobile = true,
  concourseData
}: HeaderProps) => {
  return (
    <S.Wrapper color={removeSlashesAndEmptySpaces(removeAccents(title))}>
      <S.MobileGroup>
        <Select lotteries={loterias} value={title} />
        <Logo title={title} isMobile={isMobile} />
        <Text>CONCURSO NÂº {concourseNumber}</Text>
      </S.MobileGroup>
      <S.DesktopGroup>
        <Select lotteries={loterias} value={title} />
        <Logo title={title} isMobile={false} />
        <S.TextWrapper>
          <Text>CONCURSO</Text>
          <Text size="large" weight="bold">
            {concourseNumber} - {dateFormatter(concourseData)}
          </Text>
        </S.TextWrapper>
      </S.DesktopGroup>
    </S.Wrapper>
  )
}

export default Header
