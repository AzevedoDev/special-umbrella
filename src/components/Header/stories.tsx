import { Story, Meta } from '@storybook/react/types-6-0'
import Header, { HeaderProps } from '.'

export default {
  title: 'Header',
  component: Header,
  args: {
    title: 'mega-sena',
    concourseNumber: '1020',
    loterias: [
      {
        id: 0,
        nome: 'mega-sena'
      },
      {
        id: 1,
        nome: 'quina'
      }
    ]
  },
  argTypes: {
    title: {
      control: {
        type: 'select',
        options: [
          'mega-sena',
          'quina',
          'lotof√°cil',
          'lotomania',
          'timemania',
          'dia de sorte'
        ]
      }
    }
  }
} as Meta

export const Default: Story<HeaderProps> = (args) => <Header {...args} />
