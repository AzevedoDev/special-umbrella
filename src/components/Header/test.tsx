import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Header from '.'

const loterias = [
  {
    id: 0,
    nome: 'mega-sena'
  },
  {
    id: 1,
    nome: 'quina'
  },
  {
    id: 2,
    nome: 'lotofÃ¡cil'
  },
  {
    id: 3,
    nome: 'lotomania'
  },
  {
    id: 4,
    nome: 'timemania'
  },
  {
    id: 5,
    nome: 'dia de sorte'
  }
]

const loteriaFiltrado = {
  id: '1',
  nome: 'mega-sena',
  data: '2021-05-02T01:59:06.186Z'
}

jest.mock('components/Logo', () => {
  return {
    __esModule: true,
    default: function Mock() {
      return <div data-testid="Mock Logo"></div>
    }
  }
})

jest.mock('components/Select', () => {
  return {
    __esModule: true,
    default: function Mock() {
      return <div data-testid="Mock Select"></div>
    }
  }
})

jest.mock('components/Text', () => {
  return {
    __esModule: true,
    default: function Mock() {
      return <p data-testid="Mock Text">mega-sena</p>
    }
  }
})

describe('<Header />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(
      <Header
        loterias={loterias}
        title={loteriaFiltrado.nome}
        concourseNumber={loteriaFiltrado.id}
        concourseData={loteriaFiltrado.data}
      />
    )
    expect(screen.getAllByTestId('Mock Logo')).toHaveLength(2)
    expect(screen.getAllByTestId('Mock Select')).toHaveLength(2)
    expect(screen.getAllByTestId('Mock Text')).toHaveLength(3)

    expect(screen.getAllByTestId('Mock Text')[0]).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
  it('should render correctly default props ', () => {
    renderWithTheme(
      <Header loterias={loterias} concourseData={loteriaFiltrado.data} />
    )

    expect(screen.getAllByText('mega-sena')[0]).toBeInTheDocument()
  })
})
