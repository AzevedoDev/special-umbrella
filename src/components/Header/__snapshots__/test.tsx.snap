// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render the heading 1`] = `
.c0 {
  background-color: #6BEFA3;
  height: 100%;
  width: 100%;
  font-family: 'Montserrat',-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  z-index: 10;
}

.c1 {
  display: none;
}

.c2 {
  display: none;
}

@media (min-width:768px) {
  .c0 {
    max-width: 61.5rem;
  }
}

@media (max-width:768px) {
  .c1 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    padding-top: 6.3rem;
    min-height: 45rem;
    padding-bottom: 4.1rem;
  }
}

@media (min-width:768px) {
  .c2 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    padding-top: 6.3rem;
    padding-left: 9.4rem;
    min-height: 100%;
    max-width: 61.5rem;
    padding-bottom: 9.3rem;
  }
}

<main
  class="c0"
  color="megasena"
>
  <div
    class="c1"
  >
    <div
      data-testid="Mock Select"
    />
    <div
      data-testid="Mock Logo"
    />
    <p
      data-testid="Mock Text"
    >
      mega-sena
    </p>
  </div>
  <div
    class="c2"
  >
    <div
      data-testid="Mock Select"
    />
    <div
      data-testid="Mock Logo"
    />
    <div
      class=""
    >
      <p
        data-testid="Mock Text"
      >
        mega-sena
      </p>
      <p
        data-testid="Mock Text"
      >
        mega-sena
      </p>
    </div>
  </div>
</main>
`
